lib/bump/dependency_file_parsers/python.rb

```diff
@@ -14,9 +13,59 @@ def initialize(dependency_files:)
       end
 
       def parse
-        PythonHelpers.
-          parse_requirements(@requirements.content).
-          map { |dep| Dependency.new(name: dep[0], version: dep[1]) }
+        @requirements.
+          content.
+          each_line.
+          each_with_object([]) do |line, dependencies|
+            dependency = LineParser.new(line).parse
+
+            next if dependency.nil?
+            next if dependency[:requirements].length.zero?
+
+            # Ignore dependencies with multiple requirements, since they would
+            # cause trouble at the dependency update step
+            next if dependency[:requirements].length > 1
+
+            version =
+              dependency[:requirements].first.match(LineParser::VERSION)[0]
+
+            dependencies << Dependency.new(
+              name: dependency[:name],
+              version: version,
+              language: "python"
+            )
+          end
+      end
+
+      class LineParser
+        NAME = /[a-zA-Z0-9\-_\.]+/
+        EXTRA = /[a-zA-Z0-9\-_\.]+/
+        COMPARISON = /===|==|>=|<=|<|>|~=|!=/
+        VERSION = /[a-zA-Z0-9\-_\.]+/
+        REQUIREMENT = /#{COMPARISON}\s*#{VERSION}/
+
+        REQUIREMENT_LINE =
+          /^\s*(?<name>#{NAME})
+            \s*(\[\s*(?<extras>#{EXTRA}(\s*,\s*#{EXTRA})*)\s*\])?
+            \s*(?<requirements>#{REQUIREMENT}(\s*,\s*#{REQUIREMENT})*)?
+            \s*#*\s*(?<comment>.+)?$
+          /x
```

Comment:
This is starting to look pretty scary. Looking at the [specification](https://www.python.org/dev/peps/pep-0508/#grammar) for requirements.txt, it gets a lot worse.

At this point I'd rather invest in building a general purpose, robust requirements.txt parser (ideally without regexes!) or limit the parsing we do here to the very minimum we can get away with.

Have you got examples of cases where the current approach isn't good enough (i.e. blows up or mangles the file) and is there a smaller change we can make to get it to a hygiene level?